src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1835,7 +1837,7 @@ public void terminateVirtualMachine(
         } finally {
             LOGGER.log(Level.INFO, "Clean operation starting for {0} NIC and IP", vmName);
             executionEngine.executeAsync((Callable<Void>) () -> {
-                removeIPName(resourceGroupName, vmName);
+                removeIPName(resourceGroupName, vmName, usePrivateIP);
```

Comment:
cleaner I think to pull the if statement up here?

```
if (!usePrivateIP) {
 ...
}
```

apart from that LGTM

Reply:
I didn't put it here because the removeIPName method is removing both the NIC and the public IP and I didn't want to break the NIC removal. Just wanted to stop it from attempting to remove the public IP resource when using private IP.